Homework3 with the second decision has been implemented

VitaliyBogatov opened the pull request:
Homework3

Commit:
Homework3 with the second decision has been implemented

numash commented on Practice3/Practice3.cs:
```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
```
could you please get rid of the unused dependencies?

VitaliyBogatov replied:
Done

numash commented on Practice3/Practice3.cs:
```diff
@@ -0,0 +1,46 @@
+
+namespace Practice3
+{
+    class Task
```
please, specify the access modifier for the class

VitaliyBogatov replied:
Done

numash commented on Practice3/Practice3.cs:
```diff
@@ -0,0 +1,46 @@
+            Console.Write("\n2. User defines min and max values of the range only.");
+            Console.Write("\nOption... ");
+
+            int input = Convert.ToInt16(Console.ReadLine());
```
could you please fix this exception? (happens when the user inputs a string)

![image](https://user-images.githubusercontent.com/9382934/113508417-9090c700-9558-11eb-81c4-52039980d9fb.png)

In addition, when I enter e.g. "5", the program is just closed.
Could you please handle this as well?

numash replied:
There's an exception when max value is less than min

![image](https://user-images.githubusercontent.com/9382934/113509361-cdab8800-955d-11eb-84ed-c89114a63d69.png)

AND there's no validation for negative values despite they are not natural so should not be accepted

VitaliyBogatov replied:
Done

numash commented on Practice3/Decision1.cs:
```diff
@@ -0,0 +1,62 @@
+
+namespace Practice3
+{
+    class Decision1
```
no access modifier

VitaliyBogatov replied:
Done

numash commented on Practice3/Decision1.cs:
```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
```
please, remove the unused dependencies

VitaliyBogatov replied:
Done

numash commented on Practice3/Decision2.cs:
```diff
@@ -0,0 +1,40 @@
+﻿using System;
```
please, remove the unused dependencies

VitaliyBogatov replied:
Done

numash commented on Practice3/Decision1.cs:
```diff
@@ -0,0 +1,62 @@
+
+        public int GetResult()
+        {
+            int _result = 0;
```
please, rename the local variable to just "result" (without _). 
_ - is for class fields.

VitaliyBogatov replied:
Done

numash commented on Practice3/Decision1.cs:
```diff
@@ -0,0 +1,62 @@
+
+        public void AddValues()
+        {
+            while (true)
```
please, find any way to limit this loop. we don't use infinite loops in programs/tests because it's a  week spot for a memory leaks

VitaliyBogatov replied:
I could add a counter and check that number of added values is less or equals to the countered value. But the main idea of this class was the possibility to add as many values as the user wants

numash replied:
yep, please, add a counter and set any big number so the user is not limited and the program is safe 😄

numash commented on Practice3/Decision1.cs:
```diff
@@ -0,0 +1,62 @@
+            while (true)
+            {
+                Console.WriteLine($"Type a number between {MinRange} and {MaxRange} or click enter to see results");
+                String input = Console.ReadLine();
```
I believe you wanted to use 'string' here (c# type, not String object type)

VitaliyBogatov replied:
Done. Yes, it is misprinted

numash replied:
there's still "String"

numash commented on Practice3/Decision1.cs:
```diff
@@ -0,0 +1,62 @@
+        public int MinRange { get; set; }
+        public int MaxRange { get; set; }
+
+        public void SetRange()
```
as I can see, this task is not implemented. could you please fix it?

![image](https://user-images.githubusercontent.com/9382934/113508855-085ff100-955b-11eb-9d5d-8f8e7b4ca67a.png)

VitaliyBogatov replied:
Done MaxRange must be at least MinRange+10.
However, for the decision1 a user can add as many values as s(he) wants even if the array range is between o and 1

numash commented on Practice3/Decision1.cs:
```diff
@@ -0,0 +1,62 @@
+                Console.WriteLine($"Type a number between {MinRange} and {MaxRange} or click enter to see results");
+                String input = Console.ReadLine();
+
+                if (int.TryParse(input, out int val) == true && (val >= MinRange && val <= MaxRange))
```
instead of "int.TryParse(input, out int val) == true" you should just use "int.TryParse(input, out int val)"

VitaliyBogatov replied:
Done. Absolutely agree

numash replied:
and this part is not updated either (still ==true). Maybe, the change is not pushed?

numash commented on Practice3/Decision1.cs:
```diff
@@ -0,0 +1,62 @@
+                {
+                    _list.Add(val);
+                }
+                else if (int.TryParse(input, out int j) && j < MinRange || j > MaxRange)
```
when you use several logical operators, it is better to use () to avoid human mistakes and make it easier to read

VitaliyBogatov replied:
Done

numash commented on Practice3/Decision2.cs:
```diff
@@ -0,0 +1,40 @@
+
+        public int[] SetArrayValues()
+        {
+            int[] _arr = new int[MaxRange - MinRange];
```
same with local variable naming

VitaliyBogatov replied:
Done

Commit:
Issues have been fixed
1. Unused dependencies have been removed
2. Access modifiers are added to all classes
3. String input validation
4. Switch options validation
5. Array must have at least 10 values
6. Local vriables naming

numash commented on Practice3/Decision1.cs:
```diff
@@ -0,0 +1,95 @@
+{
+    public class Decision1
+    {
+        List<int> _list = new List<int>();
```
could you please add an access modifier?

numash commented on Practice3/Decision1.cs:
```diff
@@ -0,0 +1,95 @@
+            Console.Write("Min Value: ");
+
+            bool success = false;
+            MinRange = Int32.MinValue;
```
please, change the Int32 to int everywhere
![image](https://user-images.githubusercontent.com/9382934/118265552-0413e580-b4c2-11eb-982a-664d0842094e.png)
